frame/support/src/storage/types/nmap.rs

```diff
@@ -277,7 +288,17 @@ where
 		<Self as crate::storage::StorageNMap<Key, Value>>::migrate_keys::<_>(key, hash_fns)
 	}
 
-	/// Remove all value of the storage.
+	/// Remove all values in the overlay and up to `limit` in the backend.
+	///
+	/// All values in the client overlay will be deleted, if there is some `limit` then up to
+	/// `limit` values are deleted from the client backend, if `limit` is none then all values in
+	/// the client backend are deleted.
+	///
+	/// # Note
+	///
+	/// Calling this multiple times per block with a `limit` set leads always to the same keys being
+	/// removed and the same result being returned. This happens because the keys to delete in the
+	/// overlay are not taken into account when deleting keys in the backend.
```

Comment:
Then Note is more about the underlying reason (no persistence of the applied limit in the overlay).

Wonder if renaming `limit` to `limit_backend` or `backend_fix_limit` could be a first hint.

Reply:
Actually I like your comment on sp-io better.

Reply:
> Actually I like your comment on sp-io better.

Which part exactly? Could you add some suggestions to the pr?

Reply:
The note that describe the way overlay is handled:
```
    /// The deletion would always start from `prefix` resulting in the same keys being deleted
	/// every time this function is called with the exact same arguments per block. This happens
	/// because the keys in the overlay are not taken into account when deleting keys in the
	/// backend.
```
Is more clear to me, but that may be because I know the problematic.
I was thinking a bit about it during lunch and another way to put it would be: `limit is only applied over state as it is at the start of the block, ignoring previous calls` or something like this.

Reply:
Talked to @shawntabrizi and he said we can continue with the current docs.